### PR TITLE
[NickNamer]validate bot permissions before any nick managing

### PR DESCRIPTION
**Pull request by kablekompany**
also handles exception in listener if those perms are later revoked

### CODE REVIEW

File: stickymember/__init__.py
Comment (Dav-Git) [inline stickymember/__init__.py:10]: Hello auto formatter 👋

File: nicknamer/nicknamer.py
Comment (Dav-Git) [inline nicknamer/nicknamer.py:98]: Not sure if the nickname should be removed from config here and the "freezing" be aborted. What do you think?

**Review by Dav-Git — DISMISSED** [review]
LGTM, waiting on reply to my comment before I merge.

File: nicknamer/nicknamer.py
Comment (kablekompany) [reply nicknamer/nicknamer.py:98]: That's an even better idea. I was hesitant to impose any configuration adjusting but a config clear to reduce log spam when a server for some reason removes bot perms on after freeze set up would be solid. 

Aside, I will never understand why people actually do this (remove bot perms after setting up bot)

Comment (Dav-Git) [reply nicknamer/nicknamer.py:98]: I wouldn't wipe all of config. Just the one user it failed to change. User might have gone up in the hierarchy. That would also raise discord.Forbidden :)

Comment (kablekompany) [reply nicknamer/nicknamer.py:98]: Oop touché!

**Review by Dav-Git — APPROVED** [review]

**Comment by Dav-Git** [comment]
Thanks for your contribution

**Comment by kablekompany** [comment]
A pleasure bro

**Pull request merged**
